lib/rubocop/cop/rspec/rails/persistence_called_outside_example.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module Rails
+        # Checks for persistence calls outside example blocks.
+        #
+        # Prevents persistence calls outside examples which usually run
+        # with some sort of cleanup hook. Saving outside of example causes
+        # records to leak into other tests.
+        #
+        # @ example
+        #   # bad - records created outside example group
+        #   describe User do
+        #     User.create!
+        #   end
+        #
+        #   describe User do
+        #     user = User.new
+        #     user.save!
+        #   end
+        #
+        #   # good - records created inside an example group
+        #   describe User do
+        #     it do
+        #       User.create!
+        #     end
+        #   end
+        #
+        #   describe User do
+        #     it do
+        #       user = User.new
+        #       user.save!
+        #     end
+        #   end
+        class PersistenceCalledOutsideExample < Base
+          MSG = 'Persistence called outside of example.'
+
+          # rubocop:disable Metrics/CyclomaticComplexity
+          # rubocop:disable Metrics/PerceivedComplexity
+          def on_send(node)
+            return if inside_example_scope?(node)
+            return if inside_method_definition?(node)
+            return if inside_proc_or_lambda?(node)
+            return if inside_allowed_block?(node)
+            return if allowed_receiver?(node)
+            return if allowed_method?(node)
+            return unless inside_describe_block?(node)
+            return unless persistent_call?(node)
```

Comment:
CodeClimate [was triggered by this](https://codeclimate.com/github/rubocop-hq/rubocop-rspec/pull/1016). Wondeing if it will be satisfied with
```ruby
return if ... || ... || ...
```

Reply:
I can give it a shot.